### PR TITLE
721 - add cycle and period column to the bakers table

### LINKED ISSUE #721: Bakers table update
Add `Period`, `Cycle` to the bakers table based on `Block Level`.

### PR DESCRIPTION
**Pull request by piotrkosecki**
resolves #721 
In addition I've renamed `delegate` names to `baker` in the code, as there was request in the ticket to change name of table from `delegates` to `bakers`.

### CODE REVIEW

**Comment by anonymoussprocket** [comment]
@piotrkosecki, does that mean we need to update the metadata fils?

**Comment by piotrkosecki** [comment]
> @piotrkosecki, does that mean we need to update the metadata fils?

already updated in this PR

**Comment by ivanopagano** [comment]
Could you please list the changes to db schema for infra-team to pick up? @piotrkosecki 😃

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala [hunk 1]
```diff
@@ -499,6 +499,7 @@ object TezosTypes {
       blockLevel: Int,
       timestamp: Option[Instant],
       cycle: Option[Int],
+      period: Option[Int],
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala:502]: Just wondering: are we putting too much inside this "tagging" object? Should we consider a better approach?

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala [hunk 2]
```diff
@@ -70,7 +70,7 @@ object TezosTypes {
   }
 
   /** convenience alias to simplify declarations of block hash+level tuples */
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala:72]: The comment here is terribly outdated

File: src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala [hunk 3]
```diff
@@ -890,23 +893,26 @@ object DatabaseConversions extends LazyLogging {
...
-        frozenBalance = extractBigDecimal(delegate.frozen_balance),
-        stakingBalance = extractBigDecimal(delegate.staking_balance),
-        delegatedBalance = extractBigDecimal(delegate.delegated_balance),
-        rolls = delegate.rolls.getOrElse(0),
-        deactivated = delegate.deactivated,
-        gracePeriod = delegate.grace_period,
-        blockLevel = blockLevel
-      )
+  implicit val delegateToRow =
+    new Conversion[Id, (BlockHash, Int, PublicKeyHash, Delegate, Option[Int], Option[Int]), Tables.BakersRow] {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala:897]: Thoughts: Maybe we should start assigning a type to these bunches of values?
We can probably create Value Classes if the tuple is simply wrapped, with the inconvenience of lacking named fields.

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 4]
```diff
@@ -694,12 +695,12 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
             case Some(Left(Protocol4Delegate(_, Some(pkh)))) => pkh
           }
         val taggedDelegatesKeys = taggedList.map {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/Lorre.scala:697]: delegates slipped by?

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 5]
```diff
@@ -694,12 +695,12 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
...
-          case BlockTagged(blockHash, blockLevel, timestamp, cycle, accountsMap) =>
+          case BlockTagged(blockHash, blockLevel, timestamp, cycle, period, accountsMap) =>
             import TezosTypes.Syntax._
             val delegateKeys = accountsMap.values.toList
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/Lorre.scala:700]: delegates slipped by?

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 6]
```diff
@@ -856,9 +857,9 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
-  private object DelegatesProcessor {
+  private object BakerssProcessor {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/Lorre.scala:860]: typo here

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 7]
```diff
@@ -378,7 +378,8 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
                   block.data.hash,
                   block.data.header.level,
                   Some(block.data.header.timestamp.toInstant),
-                  DatabaseConversions.extractCycle(block)
+                  DatabaseConversions.extractCycle(block),
+                  DatabaseConversions.extractPeriod(block.data.metadata)
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/Lorre.scala:382]: I'm not too fond of using what should be an internal utility for DatabaseConversions as a public method.
I'd rather extract the methods to somewhere more self-contained (TezosOptics.Blocks?) or I would create a separate method distinct from that of DatabaseConversions (which should be marked `private`)

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala (hunk 2 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala]: ok, updated

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala (hunk 4 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: yes, refactored

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala (hunk 5 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: yes, updated

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala (hunk 6 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: thanks, changed

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala (hunk 7 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: ok, moved to TezosOptics

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala (hunk 1 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala:502]: Maybe at some point when reading data from checkpoints we could just read whole block and literally `tagWithBlock` not only with the small part of it. But at this point I don't think that redesign is necessary

File: src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala (hunk 3 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala:879]: I think that could be a good idea, but as a part of a bigger refactor.

**Pull request merged**